### PR TITLE
[SERV-604] Update deps and resolve security issues

### PR DESCRIPTION
**Pull request by ksclarke**
Commit 14fec2a is the only one with substantial changes. fe8ce51 is a Checkstyle/PMD cleanup to conform to newer rules (mostly fixing missing/bad Javadocs and a few best practice things -- I used SuppressWarnings instead of trying to make substantive changes when necessary). The last (e97bb9d) is just turning on the Slack tests again, which were having temporary problems (I think related to the server, not us). Really, the only commit that needs to be reviewed, IMHO, is the first. I kept them in separate commits to isolate the substantive changes.

I left one `.snyk` exclusion unresolved because it really does need us to update to the latest version of Vert.x.

### CODE REVIEW

File: src/main/java/edu/ucla/library/bucketeer/Item.java [hunk 1]
```diff
@@ -98,7 +104,7 @@ public IFilePathPrefix getFilePathPrefix() {
      * @throws IllegalArgumentException If the value from the CSV isn't a valid state
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/bucketeer/Item.java:104]: Can the `@throws` part of the Javadoc be removed as well?

File: src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java [hunk 2]
```diff
@@ -25,7 +29,7 @@ private CodeUtils() {
      * @return A message code in integer form
      * @throws IllegalArgumentException If the supplied string isn't a message code
      */
-    public static int getInt(final String aMessageCode) throws IllegalArgumentException {
+    public static int getInt(final String aMessageCode) {
```
Comment (markmatney) [inline src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java:32]: Just making sure this change was intentional -- Javadoc still has `@throws`.

File: pom.xml [hunk 3]
```diff
@@ -1328,7 +1341,7 @@
   <parent>
     <artifactId>freelib-parent</artifactId>
     <groupId>info.freelibrary</groupId>
-    <version>6.6.3</version>
+    <version>8.0.0</version><!-- 6.6.3 -->
```
Comment (markmatney) [inline pom.xml:1344]: Is the comment at the end still needed?

File: .snyk [hunk 4]
```diff
@@ -7,54 +7,4 @@ ignore:
...
-   SNYK-JAVA-IONETTY-2812456:
-    - '*':
-        reason: We need to update to Vertx 4.x
-        expires: 2023-07-01T05:00:40.733Z
-        created: 2022-09-23T05:00:40.743Z
-   SNYK-JAVA-IONETTY-1042268:
-    - '*':
-        reason: We need to update to Vertx 4.x
-        expires: 2023-07-01T05:00:40.733Z
-        created: 2022-09-23T05:00:40.743Z
```
Comment (markmatney) [inline .snyk:19]: Were these resolved with Vert.x 3.9.14?

File: pom.xml (hunk 3 above)
Comment (ksclarke) [reply pom.xml:1344]: Duh, nope. Thanks. I'll remove.

File: src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java (hunk 2 above)
Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java:32]: Yes, PMD's rules say runtime exceptions should not be in the method's signature, but can still be in the Javadocs (I think they're still useful for the developer to know even if they shouldn't be a part of the API). Cf. https://stackoverflow.com/questions/824217/should-methods-that-throw-runtimeexception-indicate-it-in-method-signature

File: src/main/java/edu/ucla/library/bucketeer/Item.java (hunk 1 above)
Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/Item.java:104]: Cf. the other related answer below.

File: src/main/java/edu/ucla/library/bucketeer/converters/ConverterFactory.java [hunk 5]
```diff
@@ -53,7 +60,7 @@ public static Converter getConverter() {
      * @return The TIFF to JP2 converter
      * @throws KakaduNotFoundException If Kakadu has been requested but can't be found on the system
      */
-    public static Converter getConverter(final Class<?> aClass) throws KakaduNotFoundException {
+    public static Converter getConverter(final Class<?> aClass) {
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/converters/ConverterFactory.java:63]: Yes, intentional. Cf. the link below.

File: .snyk (hunk 4 above)
Comment (ksclarke) [reply .snyk:19]: Yes. Even though they're on the 4.x line, they're still doing security fixes on the 3.x line (at least until the end of the year).

File: src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java (hunk 2 above)
Comment (markmatney) [reply src/main/java/edu/ucla/library/bucketeer/utils/CodeUtils.java:32]: Ok, thanks.

**Review by markmatney — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Pull request merged**